swap_meet/vendor.py

```diff
@@ -1,2 +1,56 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if not (my_item in self.inventory and their_item in other_vendor.inventory):
+            return False
+        self.remove(my_item)
+        other_vendor.add(my_item)
+        other_vendor.remove(their_item)
+        self.add(their_item)
+        return True
+
+    def swap_first_item(self, other_vendor):
+        if not (self.inventory and other_vendor.inventory):
+            return False
+        return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+
+    def get_by_category(self, category):
+        result = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                result.append(item)
+        return result
+
+    def get_best_by_category(self, category):
+        items_category = self.get_by_category(category)
+        if not items_category:
+            return None
+
+        return max(items_category, key=lambda item: item.condition)
```

Comment:
LGTM! When we use built in functions like max and then pass a lambda as a key, it would be good practice to explain exactly what each piece of this succinct line of code is doing. It might come up in an interview and it would be a good opportunity to demonstrate your familiarity with Python, builtins, and higher order functions!